tests/test_cli.py

```diff
@@ -53,6 +55,48 @@ def raises_exception():
             raises_exception()
             mock_exit.assert_called_once_with(1)
 
+    def test_multiple_var_files_without_duplicates(self):
+        @cli.command()
+        @click.pass_context
+        def noop(ctx):
+            click.echo(ctx.obj["templating_vars"])
+
+        self.patcher_getcwd.stop()
+        with self.runner.isolated_filesystem():
+            with open("foo.yaml", "w") as foo, open("bar.yaml", "w") as bar:
+                yaml.safe_dump({"a": 1}, foo)
+                yaml.safe_dump({"b": 2}, bar)
+
+            result = self.runner.invoke(
+                cli,
+                ["--var-file", "foo.yaml", "--var-file", "bar.yaml", "noop"]
+            )
+        self.patcher_getcwd.start()
+
+        templating_vars = yaml.safe_load(result.output)
```

Comment:
this is nasty, I am echoing a dict in the noop command, and then parsing the string output back into a dict here. Open to any suggestions on how to test what the cli group is setting in the ctx object.